### PR TITLE
Fix pasted styles leaking into editor content

### PR DESCRIPTION
**Pull request by millerdev**
Happens when pasting text copied from Google Sheets.
http://manage.dimagi.com/default.asp?245729

@orangejenny @emord

### CODE REVIEW

**Comment by emord** [comment]
Is this something that happens on all ckeditor instances or is it something special about hte way we use it?

**Comment by orangejenny** [comment]
Curious why this doesn't need to strip out the `<span style="font-size:13px;...` that's in dataValue, does ckeditor take care of that?

**Comment by millerdev** [comment]
@emord asked:
> Is this something that happens on all ckeditor instances or is it something special about hte way we use it?

Good question, and I'm not sure. My guess is it has something to do with us setting `CKEDITOR.config.allowedContent = true`, which turns off ckeditor's content filter.

@orangejenny asked:
> Curious why this doesn't need to strip out the `<span style="font-size:13px;...` that's in dataValue, does ckeditor take care of that?

Yes, it removes `<span>` tags without extra help, so I didn't touch them. CKEditor dev often feels a bit like voodoo to me: poke it here and there and magic happens.

**Comment by orangejenny** [comment]
> CKEditor dev often feels a bit like voodoo to me: poke it here and there and magic happens.

Agree with that.

Regexing for HTML markup feels fragile, but I don't have a definitively better approach. Might consider doing something like `$("<div>").append(event.data.dataValue).text()` ...but that might break people pasting in legitimate HTML, if there is such a thing.

File: src/richText.js [hunk 1]
```diff
@@ -269,6 +269,11 @@ define([
             }
         });
 
+        var STYLE = /<style type="text\/css">.*?<\/style>/g;
+        editor.on('paste', function(event) {
+            event.data.dataValue = event.data.dataValue.replace(STYLE, "");
+        }, null, null, 2 );
```
Comment (orangejenny) [inline src/richText.js]: Didn't notice these arguments on first reading, what's the 2?

**Comment by millerdev** [comment]
well we at least want to preserve `<output>` tags, so I think we can't do `$("<div>").append(event.data.dataValue).text()`.

I intended to include an attribution comment, but accidentally deleted it before creating the PR. This solution was adapted from http://www.keyvan.net/2012/11/clean-up-html-on-paste-in-ckeditor/

**Comment by millerdev** [comment]
The way I'm looking at this is it fixes a very specific issue (see linked FB case) that happens when pasting from Google Sheets into Vellum, which is something our users do frequently. It's true that it is fragile, but I can't think of a better way.

File: src/richText.js (hunk 1 above)
Comment (millerdev) [inline src/richText.js]: See last comment on http://www.keyvan.net/2012/11/clean-up-html-on-paste-in-ckeditor/ (and my voodoo comment)

**Comment by orangejenny** [comment]
Yeah, that all makes sense to me.

**Pull request merged**